File: Frontend/src/domain/valueObject/body.ts
export class Body {
    private readonly value: string | null;
    constructor(input: string | null) {
        Body.validation(input);
        this.value = input; 
   }

    Value() {
        return this.value
    }

    static validation(input: string | null) {
         if (input && [...input].length > 1000) {
            throw new Error("Aboutは1000文字以下にしてください")
        }
    }
}

File: GoAPI/src/domain/valueobject/Body.go
// Package valueobject define roles of input
package valueobject

import (
	"fmt"
	"unicode/utf8"
)

type Body struct {
	body string
}

const maxbody = 1000

func NewBody(input *string) (Body, error) {
	if utf8.RuneCountInString(*input) > maxbody {
		return Body{}, fmt.Errorf("bodyは%d字以下にしてください", maxbody)
	}

	return Body{body: *input}, nil
}

func (b Body) Value() string {
	return b.body
}
